src/actions/wallet/L1/writeContractDeposit.test.ts

```diff
@@ -0,0 +1,45 @@
+import { l1CrossDomainMessengerABI } from '@eth-optimism/contracts-ts'
+import { decodeEventLog, encodeFunctionData } from 'viem'
+import { mine } from 'viem/actions'
+import { expect, test } from 'vitest'
+import { erc721ABI } from 'wagmi'
+import { accounts } from '../../../_test/constants.js'
+import { publicClient, testClient, walletClient } from '../../../_test/utils.js'
+import { base } from '../../../chains/base.js'
+import { writeContractDeposit } from './writeContractDeposit.js'
+
+test('default', async () => {
+  const functionName = 'approve'
```

Comment:
doesn't make much sense given msg.sender != msg.sender on L2

Reply:
To drive home the point that this is part of the security properties, if you couldn't approve a token from l1 then the sequencer could easily censor you from ever being able to withdraw

Reply:
@roninjin10 I understand the security properties, you are not reading the code: this is sendMessage not depositTransaction